addons/mrp/models/product.py

```diff
@@ -176,6 +188,60 @@ def _compute_used_in_bom_count(self):
         for product in self:
             product.used_in_bom_count = self.env['mrp.bom'].search_count([('bom_line_ids.product_id', '=', product.id)])
 
+    @api.depends_context('order_id')
+    def _compute_product_is_in_bom(self):
+        bom_id = self.env.context.get('order_id')
+        if not bom_id:
+            self.product_catalog_product_is_in_bom = False
+            return
+
+        read_group_data = self.env['mrp.bom.line']._read_group(
+            groupby=['product_id'],
+            aggregates=['__count'],
+        )
```

Comment:
Without any domains, those computes make no sense I think.
That said, since those computes are not used (only the search is useful), maybe instead of a compute field, you can create not-stored fields with the search method only